src/operation.ts

```diff
@@ -141,8 +155,8 @@ Operation.prototype.listenForEvents_ = function() {
  *
  * @param {function} callback
  */
-Operation.prototype.poll_ = function(callback) {
-  this.getMetadata((err, resp) => {
+Operation.prototype.poll_ = function(callback: (err?: Error|null, resp?: any) => void) {
```

Comment:
Nit: We should standardize the callback type pattern here and elsewhere. I recommend `(err: Error|null, resp?: any) => void` (though parameterizing the function on the type of resp would be better).

Reply:
I hear you.  Step 1 is to kind of identify what the various return types are across the module.  It's a little scattered and unclear.  After we're confident we have all the right types in place, I'm 100% for distilling it down to a rational set of types.